userspace/libsinsp/sinsp.h

```diff
@@ -423,35 +423,6 @@ class SINSP_PUBLIC sinsp : public capture_stats_source {
 		return m_sinsp_stats_v2;
 	}
 
-	/*!
-	  \brief Look up a thread given its tid and return its information,
-	   and optionally go dig into proc if the thread is not in the thread table.
-
-	  \param tid the ID of the thread. In case of multi-thread processes,
-	   this corresponds to the PID.
-	  \param query_os_if_not_found if true, the library will search for this
-	   thread's information in proc, use the result to create a new thread
-	   entry, and return the new entry.
-
-	  \return the \ref sinsp_threadinfo object containing full thread information
-	   and state.
-
-	  \note if you are interested in a process' information, just give this
-	  function with the PID of the process.
-
-	  @throws a sinsp_exception containing the error string is thrown in case
-	   of failure.
-	*/
-	inline const threadinfo_map_t::ptr_t& get_thread_ref(int64_t tid,
```

Comment:
I'm a bit out of the loop, so I might be missing some context, but do we have an alternative for people using the inspector to reach in and get a thread_info? Because I don't see one being added in this PR and I also don't see a way to get access to the thread_manager either...

I know the main use for Falco is to get events, but we actually do use this method in StackRox and this would break our use case.

Reply:
Never mind, I just noticed `m_thread_manager` is declared public.

![image](https://github.com/user-attachments/assets/995d993c-0af8-48ea-8fed-d26383157324)

Reply:
Yes, it is available publicly. The goal here is to remove the big amount of APIs provided by sinsp as a mirror of internal components APIs. We are doing this is order to make sinsp more maintanable and reduce the mental burden while making changes.